lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -82,8 +83,10 @@ If ($state -eq "absent") {
         If (Get-Command -Name $src -Type Application -ErrorAction SilentlyContinue) {
             $src = (Get-Command -Name $src -Type Application).Definition
         }
-        If (-not (Split-Path -Path $src -IsAbsolute)) {
-            Fail-Json -obj $result -message "Source '$src' is not found in PATH and not an absolute path."
+        If (-not (Test-Path -Path $src -IsValid)) {
```

Comment:
shouldn't this be `if (Test-Path -Path $src -IsValid)` or am I missing something. It seems you want to check if the path is absolute if it isn't a URL?

Reply:
So, if it is a url, -IsValid returns False, and only if it isn't absolute we throw the error. So I think that code is correct.
Maybe what we need are more integration tests :-)